### DISCUSSION TITLE
Performance issues under large-scale

### DISCUSSION THREAD

**Discussion by Lydanne** [mikro-orm/mikro-orm#2923] [Q&A]
**Describe the bug**
I didn't use the count method, but he executed the command, and the query was still slow.

I want to know why.

Why not use `db.User. count()`?

**Stack trace**
{ "op" : "command", "ns" : "qxg.User", "command" : { "aggregate" : "User", "pipeline" : [ { "$match" : {  } }, { "$group" : { "_id" : 1, "n" : { "$sum" : 1 } } } ], "cursor" : {  }, "lsid" : { "id" : UUID("e1b53a94-8acc-4973-9e21-e044009fa8d3") }, "$db" : "qxg" }, "keysExamined" : 0, "docsExamined" : 104561, "cursorExhausted" : true, "numYield" : 105, "nreturned" : 1, "locks" : { "ReplicationStateTransition" : { "acquireCount" : { "w" : NumberLong(1) } }, "Global" : { "acquireCount" : { "r" : NumberLong(108) } }, "Mutex" : { "acquireCount" : { "r" : NumberLong(2) } } }, "flowControl" : {  }, "storage" : { "data" : { "bytesRead" : NumberLong(6446414), "timeReadingMicros" : NumberLong(213104) } }, "responseLength" : 121, "protocol" : "op_msg", "millis" : 441, "planSummary" : "COLLSCAN", "ts" : ISODate("2022-03-15T12:16:08.402Z"), "client" : "212xx", "allUsers" : [ { "user" : "xxx", "db" : "qxg" } ], "user" : "xxx" }

**Versions**

| Dependency | Version |
| - | - |
| node | 16 |
| typescript | ? |
| mikro-orm | 4 laster |
| your-driver | mongo |

**Comment by Lydanne** [comment]
I found that the problem lies in the dependency of mongodb

**Reply by B4nan** [reply]
Please dont open issues like this
- shady problem without any description
- no reproduction at all
- using v4 which won't get any updates even if its a bug

There is no place where the ORM would do any aggregation, so "he executed the command" seems more than just impossible, if you see issues with aggregations, it is _you_ who triggered it.
